Add text on professionalism (#162)
lambdaclass/lambdaclass_hacking_learning_path: README.md

- Deliver high-quality work and taking ownership of responsibilities.
- Safeguard confidential information — never share or sell it to competitors. If you're unsure whether something can be disclosed, always check with your manager first.
- Respecting intellectual property, crediting others' contributions, and protecting confidential information—never sharing or selling it to competitors.
- Stealing, in any way, is explicitly forbidden and is grounds for immediate dismissal. This include stealing software from other projects, open source or not, without attribution or credit.
- Times are changing fast, and we always encourage trying out new tools, but at Lambda, using code provided by AI or LLMs is stricly forbidden. You can still use LLMs for other purposes, but copilot-style integrations which write code for you are a net loss, as they are not accountable regarding where they get their inspiration from and do not understand code licensing issues.

As all things, this is learned over time, with experience, and from others.
If you are ever in doubt, ask your managers, tech leads, and mentors.

> So what _is_ professionalism?

Reviewer: We explain what is professionalism but don't say a word about ethics. Should we have some points about ethics or are these points also about ethics? If it's the latter we should be more clear in the text.

Author: We should probably add a few points about ethics in a followup PR.

A professional attitude makes you trustworthy and reliable. Clients and colleagues prefer to work with people who are polite and dependable.
When you are known as somebody who is reliable, you’re more likely to be given added responsibilities and opportunities, and can make faster progress in your career. 

Some of the most professional traits:

- Self-awareness of your behavior: the most important part of creating a professional reputation is showing you behave well when at work. Every workplace has different rules, guidelines and culture, and may be more or less relaxed, but some tips that are universally applicable are:
  - Be aware of others' personal boundaries.
  - Remain calm and unemotional. 
  - Be punctual when coming to work, arriving at meetings or joining calls, to respect other's time. 
- Excellent communication: misunderstandings lead to lost productivity, so a professional always makes the effort to be understood.
- Honesty: Be open and honest, even during difficult conversations. Being able to confront people directly, rather than share your grievances about them with other people, is important in maintaining a professional attitude. 

> - Integrity: Always tell the truth. Personal accountability is owning the outcomes of your decisions, and closely related to honesty and integrity. Although you should try to do your best, everyone makes mistakes. When you do, take responsibility and offer ways to fix errors and prevent them in the future will help colleagues and supervisors maintain their trust in you. 

Reviewer: ```suggestion
- Integrity: Always tell the truth. Personal accountability is owning the outcomes of your decisions, and closely related to honesty and integrity. Although you should try to do your best, everyone makes mistakes. When you do, taking responsibility and offering ways to fix errors and prevent them in the future will help colleagues and supervisors maintain their trust in you. 
```

> - Be respectful to others: treating others with respect and kindness is part of getting your ideas across, regardless of who you are interacting with. You don't have to like everyone, or be liked by everyone, but you do have to treat them well, no matter what their role is. Respecting others also includes supporting them when they need it and being generally helpful, since thus furthers the group's goals, not individuals'. 

Reviewer: ```suggestion
- Be respectful to others: treating others with respect and kindness is part of getting your ideas across, regardless of who you are interacting with. You don't have to like everyone, or be liked by everyone, but you do have to treat them well, no matter what their role is. Respecting others also includes supporting them when they need it and being generally helpful, since this furthers the group's goals, not individuals'. 
```

- Be a dependable team player: Come prepared and plan ahead. Build a reputation as someone who keeps their word and can be trusted to deliver on commitments and deadlines. Failing to follow through can have serious consequences for those who depend on your work. When things don’t go as planned, don’t make excuses—take responsibility, do your best to get tasks back on track, and communicate early if you need additional support or resources.
- Flexibility: Commitment to your role also means being adaptable when circumstances change. As projects move forward, plans, goals, and requirements may shift, and you are expected to maintain professionalism throughout. At times, this may require adjusting your schedule—such as staying late to ensure a task or deadline is met.

> - Competency and a strong work ethic: Being highly skilled may be the first thing one thinks of of a professional. To be dependable one first needs to be capable, so as not to leave others picking up the slack or doing more than their share of the work. Over time, it is also expected that keeping knowledge and skill up to date is also expectd of a professional. No amount of highly specialized knowledge will compensate for this. Check your work for any errors before submitting it.

Reviewer: ```suggestion
- Competency and a strong work ethic: Being highly skilled may be the first thing one thinks of a professional. To be dependable one first needs to be capable, so as not to leave others picking up the slack or doing more than their share of the work. Over time, keeping knowledge and skill up to date is also expected of a professional. No amount of highly specialized knowledge will compensate for this. Check your work for any errors before submitting it.
```
